src/app/external-cluster-wizard/steps/external-cluster/provider/aks/component.ts

```diff
@@ -187,8 +188,10 @@ export class AKSClusterSettingsComponent
       this.control(Controls.Location).clearValidators();
       this.control(Controls.NodeResourceGroup).clearValidators();
       this._getAKSVmSizesForMachineDeployment(this.cluster.spec.aksclusterSpec.location).subscribe(
-        (vmSizes: string[]) => {
-          this.vmSizes = vmSizes;
+        (vmSizes: AKSVMSize[]) => {
+          if (vmSizes?.length) {
+            this.vmSizes = vmSizes.map((vmSize: AKSVMSize) => vmSize.name);
+          }
```

Comment:
```suggestion
        (vmSizes: AKSVMSize[]) => {
            this.vmSizes = vmSizes.map((vmSize: AKSVMSize) => vmSize.name);
```

In case `vmSizes` is empty we should also update `this.vmSizes`